docs/tr/docs/history-design-future.md

```diff
@@ -0,0 +1,79 @@
+# Geçmişi, Tasarımı ve Geleceği
+
+Bir süre önce, <a href="https://github.com/tiangolo/fastapi/issues/3#issuecomment-454956920" class="external-link" target="_blank">bir **FastAPI** kullanıcısı sordu</a>:
+
+> Bu projenin geçmişi nedir? Birkaç hafta içinde hiçbir yerden harika bir şeye dönüşmüş gibi görünüyor [...]
+
+İşte o geçmişin bir kısmı.
+
+## Alternatifler
+
+Bir süredir karmaşık gereksinimlere sahip API'lar oluşturuyor (Makine Öğrenimi, dağıtık sistemler, asenkron işler, NoSQL veritabanları vb.) ve farklı geliştirici ekiplerini yönetiyorum.
+
+Bu süreçte birçok alternatifi araştırmak, test etmek ve kullanmak zorunda kaldım.
+
+**FastAPI**'ın geçmişi, büyük ölçüde öncüllerinin geçmişini kapsıyor.
+
+[Alternatifler](alternatives.md){.internal-link target=_blank} bölümünde belirtildiği gibi:
+
+<blockquote markdown="1">
+
+Başkalarının daha önceki çalışmaları olmasaydı, **FastAPI** var olmazdı.
+
+Geçmişte oluşturulan pek çok araç **FastAPI**'a ilham kaynağı olmuştur.
+
+Yıllardır yeni bir framework oluşturmaktan kaçınıyordum. Başlangıçta **FastAPI**'ın çözdüğü sorunları çözebilmek için pek çok farklı framework, <abbr title="Eklenti: Plug-In">eklenti</abbr> ve araç kullanmayı denedim.
+
+Ancak bir noktada, geçmişteki diğer araçlardan en iyi fikirleri alarak bütün bu çözümleri kapsayan, ayrıca bütün bunları Python'ın daha önce mevcut olmayan özelliklerini (Python 3.6+ ile gelen <abbr title="Tip belirteçleri: Type Hints">tip belirteçleri</abbr>) kullanarak yapan bir şey üretmekten başka bir seçenek kalmamıştı.
+
+</blockquote>
+
+## Araştırma
+
+Önceki alternatifleri kullanarak, hepsinden öğrenme şansım oldu, fikirleri alıp, kendim ve çalıştığım geliştirici ekipleri için en iyi şekilde birleştirebildim.
+
+Mesela, ideal olarak standart Python tip belirteçlerine dayanması gerektiği açıktı.
+
+Ayrıca, en iyi yaklaşım zaten mevcut olan standartları kullanmaktı.
+
+Sonuç olarak, **FastAPI**'ı kodlamaya başlamadan önce birkaç ay boyunca, OpenAPI, JSON Schema, OAuth2 ve benzerlerinin tanımlamalarını inceledim. İlişkilerini, örtüşdükleri noktaları ve farklılıklarını anlamaya çalıştım.
+
+## Tasarım
+
+Sonrasında, (**FastAPI** kullanan bir geliştirici olarak) sahip olmak istediğim "API"ı tasarlamak için biraz zaman harcadım.
+
+Çeşitli fikirleri en popüler Python editörlerinde test ettim: PyCharm, VS Code, Jedi tabanlı editörler.
+
+En son <a href="https://www.jetbrains.com/research/python-developers-survey-2018/#development-tools" class="external-link" target="_blank">Python Developer Survey</a>'ine göre, kullanıcıların yaklaşık %80'inin kullandığı editörler.
+
+Bu demek oluyor ki **FastAPI**, Python geliştiricilerinin %80'inin kullandığı editörlerle test edildi. Ve diğer editörlerin çoğu benzer şekilde çalıştığından, bütün artıları neredeyse tüm editörlerde çalışacaktır.
```

Comment:
According to the change that I prefer on the previous comment, the beginning can change to "Bu da demek oluyor ki..."

Reply:
Thank you for the review. I recommend you make a "suggestion" block for these "obvious" mistakes.

Take a look at this link: [How can I manually add suggestions in code reviews on GitHub? - Stack Overflow](https://stackoverflow.com/questions/54239733/)

Thank you for this catch, I'll update it!